FindmallocMC.cmake

```diff
@@ -97,44 +102,47 @@ if(mallocMC_ROOT_DIR)
     # mallocMC variables ########################################################
     #
     set(mallocMC_VERSION "${mallocMC_VERSION_MAJOR}.${mallocMC_VERSION_MINOR}.${mallocMC_VERSION_PATCH}")
-    set(mallocMC_INCLUDE_DIRS ${mallocMC_ROOT_DIR}/include)
+    set(mallocMC_INCLUDE_DIRS ${mallocMC_ROOT_DIR}/include CACHE PATH "mallocMC include pathes")
 
     # check additional components ###############################################
     #
     foreach(COMPONENT ${mallocMC_FIND_COMPONENTS})
-        set(mallocMC_${COMPONENT}_FOUND TRUE)
+        set(mallocMC_${COMPONENT}_FOUND TRUE CACHE BOOL
+            "The mallocMC component ${COMPONENT} exists in the found installation.")
 
         if(${COMPONENT} STREQUAL "halloc")
 
             # halloc linked library #################################################
             #
             list(APPEND mallocMC_REQUIRED_VARS_LIST mallocMC_LIBRARIES)
-            find_library(${COMPONENT}_LIBRARY
+            find_library(mallocMC_${COMPONENT}_LIBRARY
                 NAMES libhalloc.a
                 PATHS "${mallocMC_ROOT_DIR}/../halloc/" ENV HALLOC_ROOT
                 PATH_SUFFIXES "lib" "bin"
-                )
-            if(${COMPONENT}_LIBRARY)
-                list(APPEND mallocMC_LIBRARIES ${${COMPONENT}_LIBRARY})
-            else(${COMPONENT}_LIBRARY)
+                DOC "Libraries for the mallocMC component ${COMPONENT}."
+            )
+            if(mallocMC_${COMPONENT}_LIBRARY)
+                list(APPEND mallocMC_LIBRARIES ${mallocMC_${COMPONENT}_LIBRARY})
+            else(mallocMC_${COMPONENT}_LIBRARY)
                 if(mallocMC_FIND_REQUIRED OR NOT mallocMC_FIND_QUIETLY)
                     message(WARNING "libhalloc.a not found. Ensure it is compiled correctly and set HALLOC_ROOT")
                 endif()
-                set(mallocMC_${COMPONENT}_FOUND FALSE)
-            endif(${COMPONENT}_LIBRARY)
+                unset(mallocMC_${COMPONENT}_FOUND CACHE)
+            endif(mallocMC_${COMPONENT}_LIBRARY)
 
             # halloc headers ########################################################
             #
-            find_path(${COMPONENT}_INCLUDE_DIR
+            find_path(mallocMC_${COMPONENT}_INCLUDE_DIR
                 NAMES halloc.h
                 PATHS "${mallocMC_ROOT_DIR}/../halloc/" ENV HALLOC_ROOT
```

Comment:
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

All right sounds very reasonable, then we go with that.
-----BEGIN PGP SIGNATURE-----
Version: APG v1.1.1

iQJCBAEBCgAsBQJVtUDEJRxBeGVsIEh1ZWJsIDxheGVsLmh1ZWJsQHBsYXNtYS5u
aW5qYT4ACgkQRy0sNk4h2UtzXQ//dUEDTA4sGuzXGUrxB+0/TtLD3aqKavEifIin
THRCJWXRjDJ+hJFDuKpy/QsY99vEA8B9TBNzDfXZYAb7L7CedWLhyUAjJ+LSkb0m
xePLKLKYAShXObEmXA6QQDFavt9D3cUu0mij3ww7cio2pwPPB5Jus37CDZaDMN5I
m9OQfnZFJc5H+gbDpo9CfjhTPYSk78r88GvdrXi5gPDkbwL6DxJ8UkYzVyV0uxfW
PsA/U/Ocei/vxL+L+zbK3KuafxURamRVbNTC1mrizOEnYpZWOIywxrTHLMZbX4Hc
8lecwOvRtX5rD5/oujxoGdtOfEY1Lz+8c2ZvLL/h7YTrFAXfq3q6jzrFgwVz8nkh
Qly6Kheo+QtKNzTfBcatcfk5metZENk6zg2tsVUjOjAkgNsAcDU7HuhtAHbMOWzH
RXTCJN1ovBg2Ca//7n70u+bVm7EnCaw4nFP0p5aVye47SpUnhrpQ/avmxYEa+Y2r
EBuLITFDd3JzviDDeKHkfjm7ZvndK32o1bRX5WrDtsDDyFOb7mP5QkveLCf/0sbE
8dhN2kF2rG2/sy5ct/c9Ty/IIrmRNhiPKy7tD/EYzjUuq3FN40+yH7PdEu3HyYut
HICNPhaDMTXRu4qjLNE3+i3o5Q2CwHqV8rnbTm9nUSZ760y14k6gOwdN+Bvs+vd3
T550blw=
=JDcC
-----END PGP SIGNATURE-----